### PR TITLE
Issue 2124 bug

### LINKED ISSUE #2101: [Bug]: Running an experiment simulation changes future simulations outputs
### PyBaMM Version

22.3

### Python Version

3.9.12

### Describe the bug

Using a DFN model, if two identical constant current discharge simulations are solved (sim1 and sim3), each of which produces the same output.

But, if the same (identical) simulations are solved but a separate simulation is run in-between (sim2) using an experiment definition, then the two solutions of sim1 and sim3 are different.

The solutions of sim1 and sim3 are equal  if any of the following are true:
a) the simulation is pre-defined (e.g. sim = pybamm.Simulation(...)) and then before and after the experiment the solutions are produced with sim1 = sim.solve
b) The in-between simulation (sim2) is not defined with an experiment (e.g. a current function or defined C_rate)

### Steps to Reproduce

import pybamm
import numpy as np
pybamm.set_logging_level(100)

model = pybamm.lithium_ion.DFN()
params = pybamm.ParameterValues('Ramadass2004')

t = np.arange(0, 7200, 0.5)
solver = pybamm.CasadiSolver(atol=1e-6, rtol=1e-6, root_method='lm', root_tol=1e-6, dt_max=5)
#%%
sol1 = pybamm.Simulation(model, parameter_values=params).solve(t, solver=solver, initial_soc=1)
sol3 = pybamm.Simulation(model, parameter_values=params).solve(t, solver=solver, initial_soc=1)

print('These are the same')
print(f"sol1 final voltage = {sol1['Terminal voltage [V]'].data[-1]:.5f} V")
print(f"sol3 final voltage = {sol3['Terminal voltage [V]'].data[-1]:.5f} V")

sol1 = pybamm.Simulation(model, parameter_values=params).solve(t, solver=solver, initial_soc=1)

sol2 = pybamm.Simulation(model, parameter_values=params, experiment=pybamm.Experiment([
    'Discharge at 1 C until 2.8 V'
])).solve(solver=solver)

sol3 = pybamm.Simulation(model, parameter_values=params).solve(t, solver=solver, initial_soc=1)

print('One of these things is not like the other one')
print(f"sol1 final voltage = {sol1['Terminal voltage [V]'].data[-1]:.5f} V")
print(f"sol3 final voltage = {sol3['Terminal voltage [V]'].data[-1]:.5f} V")

### Relevant log output

```shell
These are the same
sol1 final voltage = 2.80000 V
sol3 final voltage = 2.80000 V

One of these things is not like the other one
sol1 final voltage = 2.80000 V
sol3 final voltage = 2.77431 V
```

### LINKED ISSUE #2124: Error while simulating cycling
### PyBaMM Version

22.5

### Python Version

3.8.10

### Describe the bug

I'm struggling with fast-charge (2C) aging (SEI+irreversible plating) simulation for long cycling like 50 cycles or above. I tried different things like:
- Different models: DFN, SPMe
- Different ambient temperatures: 0-25 C
- Different i0s for SEI and plating
- Different solver modes and dt_max: Only Casadi "safe" works

The problem has two parts: first, I could run it in COMSOL with more aggressive parameters e.g. higher anode tortuosity, lower temperatures, higher charge C-rates, etc.; however, in Pybamm, I am getting the error concentration reached zero with much smaller tortuosity. 
Second, when I use smaller anode tortuosity it works without the irreversible Li plating submodel only if I include CV discharge at 2.5V at the end, otherwise, it raises the error "Experiment is infeasible" in the middle of cycling and stops at cycle 26 (reversible plating also works fine in this case). So, the problem comes from the irreversible Li plating model. I use the O'Kane model and parameters but it seems that Pybamm has some issues with it although LLI is not big at the end. The script and error for 25C are below:

### Steps to Reproduce

```
# Solving for 2C fast charge

parameter_values["Initial concentration in negative electrode [mol.m-3]"] = 300
parameter_values["Initial concentration in positive electrode [mol.m-3]"] = 49000

model = pybamm.lithium_ion.DFN({"thermal": "x-full", "SEI": "solvent-diffusion limited","lithium plating":"irreversible"})

solver= pybamm.CasadiSolver("safe")

experiment = pybamm.Experiment(
    [
        
            "Charge at 2 C until 4.2 V",
            "Hold at 4.2V until C/20",
            "Rest for 15 minutes",
            "Discharge at 0.5 C until 2.5 V",
            "Hold at 2.5 V until C/20",
            "Rest for 15 minutes"
        
    ]
    * 50)

var_pts = {
    "x_n": 40,  # negative electrode
    "x_s": 5,  # separator 
    "x_p": 10,  # positive electrode
    "r_n": 20,  # negative particle
    "r_p": 10,  # positive particle
}

sim = pybamm.Simulation(model, var_pts=var_pts, experiment=experiment, parameter_values=parameter_values, solver=solver)

sim.solve(calc_esoh=False)
sim.plot();

##################
At t = 0.00934147 and h = 1.58982e-018, the corrector convergence failed repeatedly or with |h| = hmin.
At t = 0.00949702, , mxstep steps taken before reaching tout.
At t = 0.00854179 and h = 6.02689e-026, the corrector convergence failed repeatedly or with |h| = hmin.
The linesearch algorithm failed with too small a step.
2022-06-26 22:07:10.050 - [WARNING] callbacks.on_experiment_infeasible(237): 

	Experiment is infeasible: 'final time' was triggered during 'Discharge at 0.5 C until 2.5 V'. The returned solution only contains the first 153 cycles. Try reducing the current, shortening the time interval, or reducing the period.
``` 

### Relevant log output

_No response_

### LINKED ISSUE #2138: Raise error or warning when one of the state variables gets too large
We should identify if a state variable gets too large (e.g. 1e10) as this indicates that either the model has been badly non-dimensionalized, or the model is wrong, or the parameters are wrong. This would give a clearer error message for cases like #2124 
Should this be an error or warning? Maybe a logger error (doesn't cause the code to fail)?

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Some improvements to diagnosing solver errors:
- Adds event for the particle cracking submodel when the crack is larger than the particle radius
- Adds logger error when a state vector exceeds 1e5

Fixes #2124 
Fixes #2138 
Fixes #2101 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
...
 
+## Optimizations
+
+-   Added error for when solution vector gets too large, to help debug solver errors ([#2138](https://github.com/pybamm-team/PyBaMM/pull/2138))
+
+## Bug fixes
+
+-   Fixed bug causing cut-off voltage to change after setting up a simulation with a model ([#2138](https://github.com/pybamm-team/PyBaMM/pull/2138))
+-   A single solution cycle can now be used as a starting solution for a simulation ([#2138](https://github.com/pybamm-team/PyBaMM/pull/2138))
 ## Breaking changes
```
Comment (brosaplanella) [inline CHANGELOG.md:11]:
```suggestion
-   A single solution cycle can now be used as a starting solution for a simulation ([#2138](https://github.com/pybamm-team/PyBaMM/pull/2138))

## Breaking changes
```

File: pybamm/settings.py [hunk 2]
```diff
@@ -11,6 +11,7 @@ class Settings(object):
     _heaviside_smoothing = "exact"
     _abs_smoothing = "exact"
     max_words_in_line = 4
+    max_y_value = 1e5
```
Comment (brosaplanella) [inline pybamm/settings.py:14]: Is there a way to manually change it within a particular script like with the smoothing parameters?

**Review by brosaplanella — APPROVED** [review]
Looks good, thanks!

File: pybamm/settings.py (hunk 2 above)
Comment (tinosulzer) [reply pybamm/settings.py:14]: Yes, e.g. `pybamm.settings.max_y_value = 1e10`

**Pull request merged**